### PR TITLE
Enable multiple selections of the axes of grids to be drawn

### PR DESCRIPTION
**Pull request by garychia**
# PR Details

Originally, the grid could be only drawn in one plane (X, Y or Z).
This PR enables the grids to be drawn in multiple planes.

image
image
image
image

## Related Issue
#158

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by Eideren** [comment]
How about changing this to toggle grids for each axis instead of a radio button for each combination of axis ?
i.e.:
- [ ] x
- [x] y
- [x] z
Instead of 
- [ ] x
- [ ] y
- [ ] z
- [ ] xy
- [ ] xz
- [x] yz

**Comment by garychia** [comment]
That's a good idea. I'll give it a try.

File: sources/editor/Stride.Assets.Presentation/AssetEditors/GameEditor/ViewModels/EditorGridViewModel.cs [hunk 1]
```diff
@@ -50,19 +50,67 @@ public EditorGridViewModel([NotNull] IViewModelServiceProvider serviceProvider,
...
+            return 7;
+        }
+
         public void LoadSettings([NotNull] SceneSettingsData sceneSettings)
         {
             IsVisible = sceneSettings.GridVisible;
             Color = sceneSettings.GridColor;
             Alpha = sceneSettings.GridOpacity;
             AxisIndex = sceneSettings.GridAxisIndex;
+            AxisSelection = ConvertAxisIndexToSelection(sceneSettings.GridAxisIndex);
```
Comment (manio143) [inline sources/editor/Stride.Assets.Presentation/AssetEditors/GameEditor/ViewModels/EditorGridViewModel.cs:113]: Is the `AxisIndex` used for anything else? If not, then I would suggest replacing it completely with the `AxisSelection` and passing that to the ViewportGridGizmo. Moreover, the bit mask can be encoded in an enum:
```
[Flags]
public enum ViewportGridAxis
{
    X = 1,
    Y = 2,
    Z = 4,
}
```
Which allows you to do combined enum values like `ViewportGridAxis.X & ViewportGridAxis.Y`.

File: sources/editor/Stride.Assets.Presentation/AssetEditors/Gizmos/ViewportGridGizmo.cs [hunk 2]
```diff
@@ -167,14 +176,14 @@ private Material CreateColoredTextureMaterial(Texture texture, float textureScal
             });
         }
 
-        protected override void UpdateBase(Color3 gridColor, float alpha, int gridAxisIndex, float sceneUnit)
+        private int UpdateGrid(Color3 gridColor, float alpha, int gridAxisIndex, float sceneUnit, int gridIndex)
```
Comment (manio143) [inline sources/editor/Stride.Assets.Presentation/AssetEditors/Gizmos/ViewportGridGizmo.cs:179]: Why are you not reusing `gridAxisIndex` for `gridIndex`?

**Review by manio143 — COMMENTED** [review]
Sorry for such a late review. Got a couple comments, because currently the code feels a little over complicated, but otherwise good job!

File: sources/editor/Stride.Assets.Presentation/AssetEditors/GameEditor/ViewModels/EditorGridViewModel.cs (hunk 1 above)
Comment (garychia) [reply sources/editor/Stride.Assets.Presentation/AssetEditors/GameEditor/ViewModels/EditorGridViewModel.cs:113]: You are right. Thanks for your recommendation!

File: sources/editor/Stride.Assets.Presentation/AssetEditors/Gizmos/ViewportGridGizmo.cs (hunk 2 above)
Comment (garychia) [reply sources/editor/Stride.Assets.Presentation/AssetEditors/Gizmos/ViewportGridGizmo.cs:179]: I'll try to simplify "AxisSelection" first and then refactor this part later.

**Comment by manio143** [comment]
Ok, just to be clear - is the `originAxes` array about the colored line going along the unit axis on origin point? So in the current behaviour the flat grid has both X and Z axis. I would expect each singleton grid to have 2 axis marked, and any 2+ combination of grids to have all the axis marked. Looking at your screenshots that is not the case. Am I getting this right?

**Comment by garychia** [comment]
Actually I was not sure how origin axes should be marked in this case. My understanding is that, for example, if you select X grid axis, all the origin axes excluding X axis will be marked. So in the UpdateOriginAxes function, I simply disable the origin axes selected as the grid axes, which results in the behavior the screenshots just show.

**Comment by manio143** [comment]
I got to compile your code and first thing I noticed was that the default behavior has changed
image
Which will be unexpected for the users. I'll let you figure it out, while I looked at modifying the axis behavior in a PR onto your branch: https://github.com/garychia/stride/pull/1

**Comment by garychia** [comment]
It seems that I forgot to change the default scene setting. Thanks for pointing it out. I'll try to solve it.

**Pull request merged**